BiometricLocker/BiometricLocker.swift

```diff
@@ -9,6 +9,189 @@
+        if let applicationDidEnterBackgroundDate = self.defaults.value(forKey: Key.applicationDidEnterBackgroundDate.rawValue) as? Date {
+
+            let appWasInBackgroundForLongerThan5Minutes = Date().timeIntervalSince(applicationDidEnterBackgroundDate) > self.unlockedTimeAllowance
+
+            return appWasInBackgroundForLongerThan5Minutes
+        }
+
+        return false
+    }
+
+    public init(localizedReason: String, automaticallyLocksOnBackgroundOrQuit: Bool = true, withUnlockedTimeAllowance timeAllowance: TimeInterval = LATouchIDAuthenticationMaximumAllowableReuseDuration) {
+        self.localizedReason = localizedReason
+        
+        self.unlockedTimeAllowance = timeAllowance
+
+        if automaticallyLocksOnBackgroundOrQuit {
+            NotificationCenter.default.addObserver(forName: .UIApplicationDidEnterBackground, object: nil, queue: .main) { _ in
+                // If app goes into background, we start the clocks to lock the app.
+                self.lock()
+            }
+
+            NotificationCenter.default.addObserver(forName: .UIApplicationWillTerminate, object: nil, queue: .main) { _ in
+                // If the app is killed, lock it instantly.
+                self.lock(.now)
+            }
+        }
+    }
+
+    /**
+     Tells the biometric locker to lock the app. Defaults to the currently defined time allowance (5min by default).
+
+     - Parameter when: An enum defining when the app should lock. Now, after the time allowance, or after a custom time interval.
+     */
+    public func lock(_ when: LockingTime = .afterTimeAllowance) {
+        // if we are already deactivated, return. Otherwise you can just kill the app and try again to bypass touch ID.
+        if self.isLocked { return }
+
+        let date: Date
+        switch when {
+        case .now:
+            date = Date.distantPast
+        case .afterTimeAllowance:
+            date = Date()
+        case .afterTimeInterval(let interval):
+            // We subtract our `unlockedTimeAllowance` here, to offset when checking again
+            // inside `isLocked`, otherwise it would be up to the user to know how we implement this internally
+            // and manually calculate the time difference. 
+            date = Date(timeIntervalSinceNow: (interval - self.unlockedTimeAllowance))
+        }
+
+        self.defaults.set(date, forKey: Key.applicationDidEnterBackgroundDate.rawValue)
+        self.defaults.synchronize()
+    }
+
+    /// Unlocks the app.
+    ///
+    /// - Important: Should be called if the user session is destroyed, if your app has them, otherwise you risk having the app locked whilst there's no user, preventing people from logging in / signing up.
+    public func unlock() {
+        UserDefaults.standard.removeObject(forKey: Key.applicationDidEnterBackgroundDate.rawValue)
+    }
+
+    /// Requests that the user authenticate with biometrics. Feel free to allow a fallback, like a pincode or password screen.
+
```

Comment:
I believe with this newline the documentation here will not show up when you mouse over `authenticateWithBiometrics`. Could you please delete it?

Reply:
That's not actually the case. Quite surprising. But yes, deleted it. 👍🏽

Reply:
thank you!